core/block_dragger.js

```diff
@@ -307,6 +300,41 @@ Blockly.BlockDragger.prototype.endDrag = function(e, currentDragDeltaXY) {
   Blockly.Events.setGroup(false);
 };
 
+/**
+ * Calculates the drag delta and new location values after a block is dragged.
+ * If a block would not be moved because a drag target prevents block moves,
+ * then there is no delta, and the new location is the same as the startXY.
+ * @param {!Blockly.utils.Coordinate} currentDragDeltaXY How far the pointer has
+ *     moved from the start of the drag, in pixel units.
+ * @return {{delta: (!Blockly.utils.Coordinate|undefined), newLocation:
+ *     !Blockly.utils.Coordinate}} New location after drag. delta is in
+ *     workspace units, and is undefined if a move is prevented. newLocation is
+ *     the new coordinate where the block should end up.
+ * @protected
+ */
+Blockly.BlockDragger.prototype.getNewLocationAfterMove_ = function(
+    currentDragDeltaXY) {
+  var newValues = {};
+  var preventMove = !!this.dragTarget_ &&
+      this.dragTarget_.shouldPreventBlockMove(this.draggingBlock_);
+  if (preventMove) {
+    newValues.newLocation = this.startXY_;
+  } else {
+    newValues.delta = this.pixelsToWorkspaceUnits_(currentDragDeltaXY);
+    newValues.newLocation =
+        Blockly.utils.Coordinate.sum(this.startXY_, newValues.delta);
```

Comment:
If what you're overriding in the plugin is the new location after the move, would it work to have a `getNewLocationAfterMove_` that only contains this code? This method would then only be called if the drag target did not prevent the block move.
Otherwise, you would have to also duplicate this logic of checking the drag target (shouldPreventMove) in the plugin.

Reply:
Good call. Done.